source/common/http/conn_manager_impl.cc

```diff
@@ -1891,19 +1891,20 @@ void ConnectionManagerImpl::ActiveStreamDecoderFilter::
 
 void ConnectionManagerImpl::ActiveStreamDecoderFilter::addDownstreamWatermarkCallbacks(
```

Comment:
Please add a unit test where one upstream connection is subscribed, adds a callback, another is subscribed and make sure that both callback1 doesn't get an extra onAbove but that callback2 does get the onAbove when subscribing.